nixos/modules/services/misc/sourcehut/default.nix

```diff
@@ -94,105 +133,985 @@ in
+            default = null;
+            example = "GitLab:application id:secret";
+          };
+        };
+
+        options."builds.sr.ht" = commonServiceSettings "builds" // {
+          redis = mkOption {
+            description = "The redis connection used for the celery worker.";
+            type = types.str;
+            default = "redis://localhost:6379/3";
+          };
+          shell = mkOption {
+            description = "The shell used for ssh.";
+            type = types.str;
+            default = "runner-shell";
+          };
+        };
+
+        options."git.sr.ht" = commonServiceSettings "git" // {
+          outgoing-domain = mkOption {
+            description = "Outgoing domain.";
+            type = types.str;
+            default = "https://git.localhost.localdomain";
+          };
+          post-update-script = mkOption {
+            description = ''
+              A post-update script which is installed in every git repo.
+              This setting is propagated to newer and existing repositories.
+            '';
+            type = types.path;
+            default = "${pkgs.sourcehut.gitsrht}/bin/gitsrht-update-hook";
+            defaultText = "\${pkgs.sourcehut.gitsrht}/bin/gitsrht-update-hook";
+            # Git hooks are run relative to their repository's directory,
+            # but gitsrht-update-hook looks up ../config.ini
+            apply = p: pkgs.writeShellScript "update-hook" ''
+              test -e "''${PWD%/*}"/config.ini ||
+              ln -s ${users."sshsrht".home}/../config.ini "''${PWD%/*}"/config.ini
+              exec -a "$0" '${p}' "$@"
+            '';
+          };
+          repos = mkOption {
+            description = ''
+              Path to git repositories on disk.
+              If changing the default, you must ensure that
+              the gitsrht's user as read and write access to it.
+            '';
+            type = types.str;
+            default = "/var/lib/sourcehut/gitsrht/repos";
+          };
+          webhooks = mkOption {
+            description = "The redis connection used for the webhooks worker.";
+            type = types.str;
+            default = "redis://localhost:6379/1";
+          };
+        };
+
+        options."hg.sr.ht" = commonServiceSettings "hg" // {
+          changegroup-script = mkOption {
+            description = ''
+              A changegroup script which is installed in every mercurial repo.
+              This setting is propagated to newer and existing repositories.
+            '';
+            type = types.str;
+            default = "${cfg.python}/bin/hgsrht-hook-changegroup";
+            defaultText = "\${cfg.python}/bin/hgsrht-hook-changegroup";
+          };
+          repos = mkOption {
+            description = ''
+              Path to mercurial repositories on disk.
+              If changing the default, you must ensure that
+              the hgsrht's user as read and write access to it.
```

Comment:
> the hgsrht's user **has** read and write access to it.

Reply:
AFAICS the [chowning done by systemd on StateDirectory=](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RuntimeDirectory=) happens before `BindPaths=`are applied, hence if `repos` is changed from the default to some path outside `StateDirectory=` its ownership will be preserved.